### DISCUSSION TITLE
Gas costs of various actions

### DISCUSSION THREAD

**Discussion by lumos42** [aptos-labs/aptos-developer-discussions#324] [General]
I couldn't find any documentation of gas costs of various operations like borrow from global storage, table lookup, fungible asset mint. I understand that I can run profiling on the modules and figure out that way, but it would be nice to have some approximate gas cost of common operations to help guide the design decisions.

**Comment by vgao1996** [comment]
Hi @lumos42, thanks for the feedback! This is something we plan to build!
- The major reason why we haven't come up with a doc showing the exact gas costs is because such doc would have required a huge amount of maintenance, if done manually.
  - What's difficult is to ensure the doc is always up to date (compared to the code).
- However recently we've made some progress
  - We've figured out a way to embed custom annotations (info required for the doc) directly into the Rust code using procedural macros, ensuring atomic updates of the code and the annotations.
  - With such annotations, we can then write a tool to generate human readable gas docs automatically.

**Reply by lumos42** [reply]
Thank you fro the response! I understand how difficult it can be to maintain these docs manually, and this looks like progress in the right direction. Do you also have an estimated timeline of when these automatic docs will be ready?
